mtags/src/main/scala-2/scala/meta/internal/pc/PcCollector.scala

```diff
@@ -95,102 +95,115 @@ abstract class PcCollector[T](
     else (pos, false)
   }
 
-  def result(): List[T] = {
-
-    lazy val namedArgCache = {
-      val parsedTree = parseTree(unit.source)
-      parsedTree.collect { case arg @ AssignOrNamedArg(_, rhs) =>
-        rhs.pos.start -> arg
-      }.toMap
-    }
-    def fallbackSymbol(name: Name, pos: Position) = {
-      val context = doLocateImportContext(pos)
-      context.lookupSymbol(name, sym => sym.isType) match {
-        case LookupSucceeded(_, symbol) =>
-          Some(symbol)
-        case _ => None
-      }
+  lazy val namedArgCache = {
+    val parsedTree = parseTree(unit.source)
+    parsedTree.collect { case arg @ AssignOrNamedArg(_, rhs) =>
+      rhs.pos.start -> arg
+    }.toMap
+  }
+  def fallbackSymbol(name: Name, pos: Position): Option[Symbol] = {
+    val context = doLocateImportContext(pos)
+    context.lookupSymbol(name, sym => sym.isType) match {
+      case LookupSucceeded(_, symbol) =>
+        Some(symbol)
+      case _ => None
     }
+  }
 
-    // First identify the symbol we are at, comments identify @@ as current cursor position
-    val soughtSymbols: Option[Set[Symbol]] = typedTree match {
-      /* simple identifier:
-       * val a = val@@ue + value
-       */
-      case (id: Ident) =>
-        // might happen in type trees
-        // also this doesn't seem to be picked up by semanticdb
-        if (id.symbol == NoSymbol)
-          fallbackSymbol(id.name, pos).map(symbolAlternatives)
-        else {
-          Some(symbolAlternatives(id.symbol))
+  lazy val soughtSymbols: Option[(Set[Symbol], Position)] = typedTree match {
```

Comment:
I had to move most logic outside of `result` function, because they are useful also for `prepareRename`